server/src/main/java/org/opensearch/tracing/DefaultTracer.java

```diff
@@ -0,0 +1,213 @@
+    private static final String PARENT_SPAN_NAME = "parent_span_name";
+    private static final String ROOT_SPAN = "RootSpan";
+
+    private final ThreadPool threadPool;
+    private final ClusterService clusterService;
+    private final io.opentelemetry.api.trace.Tracer openTelemetryTracer;
+
+    public DefaultTracer(io.opentelemetry.api.trace.Tracer openTelemetryTracer, ThreadPool threadPool, ClusterService clusterService) {
+        this.openTelemetryTracer = openTelemetryTracer;
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public void startSpan(String spanName, Map<String, Object> attributes, Level level) {
+        startSpan(spanName, attributes, null, level);
+    }
+
+    @Override
+    public void startSpan(String spanName, Map<String, Object> attributes, Span parentSpan, Level level) {
+        if (parentSpan == null) {
+            parentSpan = getCurrentSpan();
+        }
+        Span span = createSpan(spanName, parentSpan, level);
+        setCurrentSpanInContext(span);
+        setSpanAttributes(span, attributes);
+    }
+
+    @Override
+    public void endSpan() {
+        Span currentSpan = getCurrentSpan();
+        if (currentSpan != null) {
+            endSpan(currentSpan);
+            setCurrentSpanInContext(currentSpan.getParentSpan());
+        }
+    }
+
+    @Override
+    public void addAttribute(String key, Object value) {
+        Span currentSpan = getCurrentSpan();
+        addSingleAttribute(key, value, currentSpan);
+    }
+
+    @Override
+    public void addEvent(String event) {
+        Span currentSpan = getCurrentSpan();
+        if (currentSpan instanceof OSSpan && ((OSSpan) currentSpan).getOtelSpan() != null) {
+            ((OSSpan) currentSpan).getOtelSpan().addEvent(event);
+        }
+    }
+
+    private Span createSpan(String spanName, Span parentSpan, Level level) {
+        return isLevelEnabled(level) ?
+                createOSSpan(spanName, parentSpan, level) :
+                createNoopSpan(spanName, parentSpan, level);
+    }
+
+    private Span createOSSpan(String spanName, Span parentSpan, Level level) {
+        OSSpan parentOSSpan = getLastOSSpanInChain(parentSpan);
+        io.opentelemetry.api.trace.Span otelSpan = createOtelSpan(spanName, parentOSSpan);
+        Span span = new OSSpan(spanName, otelSpan, parentOSSpan, level);
+        logger.info("Starting OtelSpan spanId:{} name:{}: traceId:{}", otelSpan.getSpanContext().getSpanId(),
+                span.getSpanName(), otelSpan.getSpanContext().getTraceId());
+        return span;
+    }
+
+    private NoopSpan createNoopSpan(String spanName, Span parentSpan, Level level) {
+        logger.info("Starting Noop span name:{}", spanName);
+        return new NoopSpan(spanName, parentSpan, level);
+    }
+
+    private OSSpan getLastOSSpanInChain(Span parentSpan) {
+        while (!(parentSpan instanceof OSSpan)) {
```

Comment:
In case of noop can we add an event to the parent may be that this is mapped to this parent becuase of noop span in-between. Also lets' update the documentation on how level works. May be in the Levels enum.